### ISSUE TITLE
 Rule proposal: `prefer-array-includes`

### ISSUE THREAD

**Issue by Fabrice-TIERCELIN** [sindresorhus/eslint-plugin-unicorn#1058]
Some built-ins have [`Array#includes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes) in addition to [`Array#some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some). Prefer `Array#includes()` over use of `Array#some()` passing a function that tests equality as parameter.

This rule is partly fixable.

## Fail

```js
[].some(x => x === 'foo');
values.some(x => 'foo' === x);
values.some(x => {return x === 'foo';});
['foobar'].some(x => {return 'foo' === x;});
values.some(function (x) {return x === 'foo';});
['foobar'].some(function (x) {return 'foo' === x;});
```

## Pass

```js
[].some(i => i === list[i]);
[].some(x => x === 'foo' && isValid);
[].some(x => y === 'foo');
[].some(x => y.x === 'foo');
[].some(x => x == 'foo');
['foobar'].some(function (x) {return 'foo' == x;});
[].some(x => x !== 'foo');
values.some(function (x) {return x !== 'foo';});
```

**Comment by fisker** [comment]
:+1: for this, but we already have [`prefer-includes`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-includes.md), we can add this case to the existing one.

**Comment by fisker** [comment]
We'll have funny thing,

```js
const array = [1,2,3];

if (array.filter(x => x=== 1)[0]) {
}
```

`prefer-array-find` fix to

```js
const array = [1,2,3];

if (array.find(x => x=== 1)) {
}
```

`prefer-array-some` fix to

```js
const array = [1,2,3];

if (array.some(x => x=== 1)) {
}
```

`prefer-includes`(This probosal) fix to 

```js
const array = [1,2,3];

if (array.includes(1)) {
}
```

`prefer-set-has` fix to

```js
const array = new Set([1,2,3]);

if (array.has(1)) {
}
```

LOL~~~~

**Comment by Fabrice-TIERCELIN** [comment]
I will look at it.
